token/wallet.tact

```diff
@@ -2,164 +2,116 @@
+        verify(transfer.destination.workchain() == 0, "destination must be local");
+        verify(self.owner == msg.sender, "sender must be owner");
+        verify(self.balance - transfer.amount >= 0, "not enough balance");
+
+        let fwd_count = if (transfer.forward_ton_amount != 0) { 2 } else { 1 };
+        let required_input = transfer.forward_ton_amount + 
+            msg.fwd_fee * msg.fwd_count + 
+            2 * GAS_CONSUMPTION + 
+            STORAGE_MIN_CONSUMPTION;
+        verify(msg.amount > required_input, "not enough value");
+
+        self.balance -= transfer.amount;
+
+        Jetton.init(
+            msg.destination,
+            self.master,
+            self.wallet_code,
+            code: self.wallet_code
+        ).receive_tokens(InternalTransfer {
             query_id: msg.query_id,
             amount: msg.amount,
-            from: state.owner,
+            from: self.owner,
             response_address: msg.response_destination,
             forward_ton_amount: msg.forward_ton_amount,
             forward_payload: msg.forward_payload,
         })
-        }.serialize())
-        .send_to(to_wallet_address);
-}
-
-fn receive_tokens(
-    @mut state: WalletState, 
-    msg: InternalTransfer, 
-    sender: ContractAddress, 
-    my_ton_balance: Coins, 
-    fwd_fee: Coins, 
-    msg_value: Coins
-) {
-    verify(
-        sender === state.master || 
-            sender == WalletAddress.from_data(msg.from, state.master, state.wallet_code), 
-        "sender must be master or user wallet"
-    );
-
-    let ton_balance_before_msg = my_ton_balance - msg_value;
-    let storage_fee = STORAGE_MIN_CONSUMPTION - ton_balance_before_msg.min(STORAGE_MIN_CONSUMPTION);
-    let @mut msg_value = msg_value - (storage_fee + GAS_CONSUMPTION);
-
-    if (forward_ton_amount != 0) {
-        msg_value -= (forward_ton_amount + fwd_fee);
-        let payload = msg.forward_payload;
-        RequestBuilder.new()
-            .ihr_disabled()
-            .pay_transfer_fee_separately()
-            .money(forward_ton_amount)
-            .body(TransferNotification {
-                query_id: msg.query_id,
-                amount: msg.amount,
-                from: msg.from,
-                payload: payload,
-            })
-            .send_to(owner_address);
+        .send(
+            ihr_disabled: true,
+            can_be_bounced: true,
+            carry_message_input: true,
+        );
     }
 
-    if (!response_address.is_null() && msg_value > 0) {
-        RequestBuilder.new()
-            .ihr_disabled()
-            .ignore_action_phase_errors()
-            .money(msg_value)
-            .body(Excesses {
-                query_id: msg.query_id,
-            })
-            .send_to(response_address);
+    internal receive_tokens#178d4519(transfer: InternalTransfer) {
+        verify(
+            msg.sender === self.master || 
+                sender == WalletAddress.from_data(msg.from, self.master, self.wallet_code), 
```

Comment:
Is this code up-to-date? Should we do Jetton.init(...).address() here?

Reply:
Fixed in [fe0164c](https://github.com/tact-lang/tact-bikeshed/pull/5/commits/fe0164c40cc0c43eadd7ac984fd59c6ed6c0cdd9)